gateway/testutil.go

```diff
@@ -1144,6 +1144,8 @@ func (s *Test) newGateway(genConf func(globalConf *config.Config)) *Gateway {
 
 	go s.reloadSimulation(s.ctx, gw)
 
+	gw.started = true
```

Comment:
The bool value raises some red flags (concurrent access to read it, polling rather than e.g. initCtx context?). I don't see it being used in the PR, possibly a separate PR is needed to address this. Ideally, I'd consider removing it.